src/archivematicaCommon/lib/databaseFunctions.py

```diff
@@ -229,8 +229,12 @@ def logTaskCompletedSQL(task):
     print("Logging task output to db", task.UUID)
     taskUUID = task.UUID.__str__()
     exitCode = task.results["exitCode"].__str__()
-    stdOut = task.results["stdOut"]
-    stdError = task.results["stdError"]
+    # truncate task stdout/stderror logged to the database
+    # (to prevent mysql server errors, refs. #10732) 
+    TASK_STDOUT_LIMIT = 16*1024*1024
+    TASK_STDERR_LIMIT = 16*1024*1024
+    stdOut = (task.results["stdOut"][:TASK_STDOUT_LIMIT] + '..') if len(task.results["stdOut"]) > TASK_STDOUT_LIMIT else task.results["stdOut"]
+    stdError = (task.results["stdError"][:TASK_STDERR_LIMIT] + '..') if len(task.results["stdError"]) > TASK_STDERR_LIMIT else task.results["stdError"]
```

Comment:
Yes, I think you should truncate the stdout/stderr strings *after* they are unicode. The strToUnicode func will raise a UnicodeDecodeError (unless you pass obstinate=True). You could easily raise such an exception by truncating bytes and decoding the result to Unicode, e.g., `a\u0301'.encode('utf8')[:2].decode('utf8')` would do it.